### ISSUE TITLE
following PEP8 standards

### ISSUE THREAD

**Issue by chantisnake** [WheatonCS/Lexos#509]
the wiggling line in pycharm are making me crazy.
like this:
![capture](https://cloud.githubusercontent.com/assets/10100995/22407276/47b82a6a-e65b-11e6-9b34-67e7c0b62820.PNG)

https://pypi.python.org/pypi/autopep8 seems like a great tool, also pycharm has a auto formatting.

also pycharm offers to reformat and clean your code before you commit:
![capture](https://cloud.githubusercontent.com/assets/10100995/22407303/ead35c60-e65b-11e6-9230-75d106a36875.PNG)

we have also discussed to use CI to check for PEP8 to ensure that our code is clean, readable, and standard
(https://github.com/WheatonCS/Lexos/issues/497)

**Comment by scottkleinman** [comment]
CI integreation is on the [roadmap for v3.2](https://github.com/WheatonCS/Lexos/blob/master/roadmap.md#v32-minor-enhancements-code-cleanup-and-workflow-improvements). I am very hopeful that v3.1 is now done and pushed to the master branch, will go on the new server sometime next month, and will ship automatically with your new installer. So we're close to starting on v3.2, although real development may not take place until the summer. But maybe you should use this issue to add any other recommendations (like autopep8) for when we begin this transition.

In the meantime, just think of those wiggly underlines as gentle, calming ocean waves...

**Comment by chantisnake** [comment]
I personally think the huge changes (the changes impacts deeply on future development of the software), like python3 and PEP8 should happen before we get into other stuff, because the more stuff we put into the software by now, will add more barrier to these changes.

For example, the more code we add to this repo, there will be more things to change when we want to migrate to PEP8 and Python3.

since migrating to Python3 is basically done, and PEP8 is not that hard a task, I would suggest to put these to v3.2.

What do you think?

By the way, I think we need to talk to Mark about this (he seems busy lately, since he is not on github), I guess we should email or skype?

**Comment by scottkleinman** [comment]
The roadmap is a draft, so we'll probably change some things after consultation. Right now, I've added links to issue numbers and arranged the items for each version in order of issue number, not in order of precedence.

Over the last month, we have applied for two grants that will fund work that covers some items in this roadmap, so we do need to slow down for a bit to recover and to make sure that our development plan aligns with the work plans we outlined in the grant proposals.

The changes in 3.1 are really bug fixes and minor enhancements, some of which were in place before you started the Python 3 branch, so it should be pretty easy to merge them in. But we do need to decide at what point we begin developing in Python 3. Let's take a week to get settled into the new semester, and then we can try to arrange a group consult. From what I heard from Mark, the lab is getting all new boxes, and I'm sure the setup will have its fair share of headaches.
